src/button-layout.tsx

```diff
@@ -149,8 +151,10 @@ const ButtonLayout: React.FC<ButtonLayoutProps> = ({
                         el.classList.contains(classes.link) ? 0 : (el as HTMLElement).offsetWidth + 1
                     );
                     const maxChildWidth = Math.ceil(Math.max(...childrenWidths, BUTTON_MIN_WIDTH));
-                    updateButtonWidth(maxChildWidth);
-                    updateIsMeasuring(false);
+                    ReactDom.unstable_batchedUpdates(() => {
+                        updateButtonWidth(maxChildWidth);
+                        updateIsMeasuring(false);
+                    });
```

Comment:
instead of batching updates, you could have a single state object with 2 keys

Reply:
done